src/hooks/useClientErrorHandler.ts

```diff
@@ -0,0 +1,113 @@
+import { PullResponderTrigger } from 'document-drive';
+import {
+    PullResponderTriggerData,
+    Trigger,
+} from 'document-model-libs/document-drive';
+import { useCallback, useRef, useState } from 'react';
+import { useDocumentDriveServer } from './useDocumentDriveServer';
+
+export type ClientErrorHandler = {
+    strandsErrorHandler: (
+        driveId: string,
+        trigger: Trigger,
+        status: number,
+    ) => Promise<void>;
+};
+
+const DELAY_LIMIT = 100000;
+
+export const useClientErrorHandler = (): ClientErrorHandler => {
+    const [handlingInProgress, setHandlingInProgress] = useState<string[]>([]);
+    const [pullResponderTriggerMap, setPullResponderTriggerMap] = useState<
+        Map<string, PullResponderTrigger>
+    >(new Map());
+    const { addTrigger, removeTrigger, registerNewPullResponderTrigger } =
+        useDocumentDriveServer();
+
+    const pullResponderRegisterDelay = useRef<Map<string, number>>(new Map());
+
+    const handleStrands404 = useCallback(
+        async (driveId: string, trigger: Trigger, handlerCode: string) => {
+            setHandlingInProgress(state => [...state, handlerCode]);
+
+            const triggerData =
+                trigger.data as unknown as PullResponderTriggerData;
+
+            try {
+                let pullResponderTrigger: PullResponderTrigger;
```

Comment:
Can we avoid the non null assert by doing:

```
let pullResponderTrigger = pullResponderTriggerMap.get(handlerCode);
if (!pullResponderTrigger) {
 pullResponderTrigger = await registerNewPullResponderTrigger(
 ...
}
```

Reply:
fixed at [1a96a4d](https://github.com/powerhouse-inc/connect/pull/436/commits/1a96a4d291ccb430f8f960ea7450c90dd1e0ae43)